puppet/modules/jenkins_job_builder/files/theforeman.org/pipelines/lib/foreman_infra.groovy

```diff
@@ -25,3 +25,58 @@ def set_job_build_description(job_name) {
 
     currentBuild.description += build_description
 }
+
+def runIndividualCicoJob(job_name, number = 0) {
+    sleep(number * 5) //See https://bugs.centos.org/view.php?id=14920
+    try {
+        withCredentials([string(credentialsId: 'centos-jenkins', variable: 'PASSWORD')]) {
+            runPlaybook(
+                playbook: 'ci/centos.org/ansible/jenkins_job.yml',
+                extraVars: [
+                    "jenkins_job_name": "${job_name}",
+                    "jenkins_username": "foreman",
+                    "jenkins_job_link_file": "${env.WORKSPACE}/jobs/${job_name}"
+                ],
+                sensitiveExtraVars: ["jenkins_password": "${env.PASSWORD}"]
+            )
+        }
+    } finally {
+        script {
+            set_job_build_description(job_name)
+        }
+    }
+}
+
+def runCicoJob(job_name) {
+    node('el') {
+        script {
+            git_clone_foreman_infra()
+            try {
+                runIndividualCicoJob(job_name)
+            } finally {
+                deleteDir()
+            }
+        }
+    }
+}
+
+def runCicoJobsInParallel(jobs) {
+    def branches = [:]
+    for (int i = 0; i < jobs.size(); i++) {
```

Comment:
I think this can be written as

```groovy
    jobs.eachWithIndex { job ->
        branches[job['name']] = {
            runIndividualCicoJob(job['job'], index)
        }
    }
```